test/integration/import_and_export/parquet_test.py

```diff
@@ -0,0 +1,101 @@
+from pathlib import Path
+
+import pyarrow as pa
+import pytest
+from integration.import_and_export.helper import select_result
+from pyarrow import parquet as pq
+
+
+def prepare_parquet_table(list_dict: list[dict]) -> pa.Table:
+    table = pa.Table.from_pylist(list_dict)
+    table = table.set_column(2, "REGISTER_DT", table["REGISTER_DT"].cast(pa.date32()))
+    table = table.set_column(
+        3, "LAST_VISIT_TS", table["LAST_VISIT_TS"].cast(pa.timestamp("ns"))
+    )
+    # this is not ideal as the database & Python data use bool
+    return table.set_column(4, "IS_GRADUATING", table["IS_GRADUATING"].cast(pa.int64()))
+
+
+@pytest.mark.parquet
+class TestExportToParquet:
+    @staticmethod
+    def test_export_single_file(connection, fill_table, tmp_path, table_name, all_data):
+        expected = prepare_parquet_table(all_data.list_dict)
+
+        filepath = tmp_path / "single_parquet_dir"
+        connection.export_to_parquet(dst=filepath, query_or_table=table_name)
+
+        assert len(list(filepath.glob("*"))) == 1
+        # can be a single file name or directory name
+        assert pq.read_table(filepath) == expected
+
+    @staticmethod
+    def test_export_multiple_files(
+        connection, fill_table, tmp_path, table_name, all_data, number_entries
+    ):
+        expected = prepare_parquet_table(all_data.list_dict)
+        rows_per_file = 5
+
+        filepath = tmp_path / "multiple_parquet_dir"
+        connection.export_to_parquet(
+            dst=filepath,
+            query_or_table=table_name,
+            # parquet requires the user to set max_rows_per_group <= max_rows_per_file
```

Comment:
Yes, because a file consists of one or more row groups. Row groups are "partitions" inside a file.  Each row group can be read independently of other row groups.

Reply:
Yes, sometimes I just add comments for later developers to reduce the onboarding load.

This comes from a discussion with KK and Thomas, where I used some set operations in PTB that took them quite some time to figure out.